pkg/eventfilter/subscriptionsapi/prefix_filter.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subscriptionsapi
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"go.uber.org/zap"
+	"knative.dev/pkg/logging"
+
+	"knative.dev/eventing/pkg/eventfilter"
+	"knative.dev/eventing/pkg/eventfilter/attributes"
+)
+
+type prefixFilter struct {
+	attribute, prefix string
+}
+
+// NewPrefixFilter returns an event filter which passes if the value of the context
+// attribute in the CloudEvent is prefixed with prefix.
+func NewPrefixFilter(attribute, prefix string) eventfilter.Filter {
+	return &prefixFilter{
+		attribute: attribute,
+		prefix:    prefix,
+	}
```

Comment:
nit, saving some memory.
```suggestion
	if attribute == "" || prefix == "" {
		return nil
	}
	return &prefixFilter{
		attribute: attribute,
		prefix:    prefix,
	}
```